src/main/scala/viper/gobra/frontend/PackageResolver.scala

```diff
@@ -50,68 +58,102 @@ object PackageResolver {
     for {
       // pkgDir stores the path to the directory that should contain source files belonging to the desired package
       pkgDir <- getLookupPath(n.importPath, includeDirs)
-      sourceFiles = getSourceFiles(pkgDir.toFile)
+      sourceFiles = getSourceFiles(pkgDir)
       // check whether all found source files belong to the same package (the name used in the package clause can
       // be absolutely independent of the import path)
       pkgName <- checkPackageClauses(sourceFiles, n.importPath)
+      // close all files as we do not need them anymore:
+      _ = sourceFiles.foreach(_.close())
     } yield pkgName
   }
 
-  /**
-    * Resolves importPath using includeDirs to a directory which exists and from which source files should be retrieved
-    */
-  private def getLookupPath(importPath: String, includeDirs: Vector[File]): Either[String, Path] = {
+  private def getIncludeResources(includeDirs: Vector[File]): Vector[InputResource] = {
+    includeDirs.map(FileResource)
+  }
+
+  private lazy val getStubResources: Vector[InputResource] = {
+    // get path to stubs
+    stubDirectories.flatMap(stubDir => {
+      val nullableResourceUri = getClass.getClassLoader.getResource(stubDir).toURI
+      for {
+        resourceUri <- Option.when(nullableResourceUri != null)(nullableResourceUri)
+        resource <- resourceUri.getScheme match {
+          case s if s == fileUriScheme => Some(FileResource(Paths.get(resourceUri).toFile))
+          case s if s == jarUriScheme => Some(BaseJarResource(resourceUri, stubDir))
```

Comment:
I did not get why there are two possible Uri schemes when you get a stub resource

Reply:
When you run Gobra's tests then files in the stub folder are regular files in your filesystem. However, when you invoke Gobra with `sbt run -i ...`, sbt seems to pack everything into a jar first and execute the jar. The stub files also get packed into the jar. As the jar is some kind of zip file, accessing the stubs as regular files (e.g. calling `toFile` on a `Path`) fails. I have found two ways how to access such "packed files" in a jar: (1) you specify the path to such a file and read it using streams or (2) you create a new "filesystem" for the files in the jar. I went with option 2 as this allows scanning the directory structure as I do not make any assumption about which files are present or not.

Reply:
That makes sense. Thanks!